File: 0217.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> nums2 = new HashSet<>();
        for(int i = 0; i < nums.length; i++){
            if(nums2.contains(nums[i])){
                return true;
            }
            nums2.add(nums[i]);
        }
        return false;
    }
}

File: snap_prep/0217.py
class Solution:
    '''
    def containsDuplicate(self, nums: List[int]) -> bool:
        if not nums:
            return False
        nums2 = list(set(nums))
        nums.sort()
        nums2.sort()
        return not nums == nums2
    '''
    
    def containsDuplicate(self, nums: List[int]) -> bool:
        nums2 = set()
        for n in nums:
            if n in nums2:
                return True
            nums2.add(n)
        return False
